### ISSUE TITLE
New @polkadot/types bundle throws errors when running tsc

### ISSUE THREAD

**Issue by joelamouche** [polkadot-js/api#4216]
In the moonbeam/moonbeam-types-bundle folder, when running `npm run build`, I get:

```
node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:280:57 - error TS2304: Cannot find name 'XcmV2TraitsError'.

280         readonly asExecutionResult: Option<ITuple<[u32, XcmV2TraitsError]>>;
                                                            ~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:434:27 - error TS2304: Cannot find name 'SpFinalityGrandpaAppPublic'.

434         readonly grandpa: SpFinalityGrandpaAppPublic;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:435:24 - error TS2304: Cannot find name 'SpConsensusBabeAppPublic'.

435         readonly babe: SpConsensusBabeAppPublic;
                           ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:436:28 - error TS2304: Cannot find name 'PalletImOnlineSr25519AppSr25519Public'.

436         readonly imOnline: PalletImOnlineSr25519AppSr25519Public;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:437:33 - error TS2304: Cannot find name 'PolkadotPrimitivesV0ValidatorAppPublic'.

437         readonly paraValidator: PolkadotPrimitivesV0ValidatorAppPublic;
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:438:34 - error TS2304: Cannot find name 'PolkadotPrimitivesV1AssignmentAppPublic'.

438         readonly paraAssignment: PolkadotPrimitivesV1AssignmentAppPublic;
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:439:38 - error TS2304: Cannot find name 'SpAuthorityDiscoveryAppPublic'.

439         readonly authorityDiscovery: SpAuthorityDiscoveryAppPublic;
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:444:28 - error TS2304: Cannot find name 'FrameSystemRawOrigin'.

444         readonly asSystem: FrameSystemRawOrigin;
                               ~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:447:29 - error TS2304: Cannot find name 'PalletCollectiveRawOrigin'.

447         readonly asCouncil: PalletCollectiveRawOrigin;
                                ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:449:40 - error TS2304: Cannot find name 'PalletCollectiveRawOrigin'.

449         readonly asTechnicalCommittee: PalletCollectiveRawOrigin;
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-kusama.d.ts:451:38 - error TS2304: Cannot find name 'PolkadotRuntimeParachainsOriginPalletOrigin'.

451         readonly asParachainsOrigin: PolkadotRuntimeParachainsOriginPalletOrigin;
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:33:61 - error TS2304: Cannot find name 'SpCoreSr25519Public'.

33     interface PolkadotPrimitivesV0CollatorAppPublic extends SpCoreSr25519Public {
                                                               ~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:36:64 - error TS2304: Cannot find name 'SpCoreSr25519Signature'.

36     interface PolkadotPrimitivesV0CollatorAppSignature extends SpCoreSr25519Signature {
                                                                  ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:85:27 - error TS2304: Cannot find name 'SpFinalityGrandpaAppPublic'.

85         readonly grandpa: SpFinalityGrandpaAppPublic;
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:86:24 - error TS2304: Cannot find name 'SpConsensusBabeAppPublic'.

86         readonly babe: SpConsensusBabeAppPublic;
                          ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:87:28 - error TS2304: Cannot find name 'PalletImOnlineSr25519AppSr25519Public'.

87         readonly imOnline: PalletImOnlineSr25519AppSr25519Public;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:90:38 - error TS2304: Cannot find name 'SpAuthorityDiscoveryAppPublic'.

90         readonly authorityDiscovery: SpAuthorityDiscoveryAppPublic;
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:93:62 - error TS2304: Cannot find name 'SpCoreSr25519Public'.

93     interface PolkadotPrimitivesV0ValidatorAppPublic extends SpCoreSr25519Public {
                                                                ~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:96:63 - error TS2304: Cannot find name 'SpCoreSr25519Public'.

96     interface PolkadotPrimitivesV1AssignmentAppPublic extends SpCoreSr25519Public {
                                                                 ~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:109:28 - error TS2304: Cannot find name 'FrameSystemRawOrigin'.

109         readonly asSystem: FrameSystemRawOrigin;
                               ~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:112:29 - error TS2304: Cannot find name 'PalletCollectiveRawOrigin'.

112         readonly asCouncil: PalletCollectiveRawOrigin;
                                ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:114:40 - error TS2304: Cannot find name 'PalletCollectiveRawOrigin'.

114         readonly asTechnicalCommittee: PalletCollectiveRawOrigin;
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:147:32 - error TS2304: Cannot find name 'SpRuntimeHeader'.

147         readonly parentHeader: SpRuntimeHeader;
                                   ~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:158:65 - error TS2304: Cannot find name 'SpCoreSr25519Signature'.

158     interface PolkadotPrimitivesV0ValidatorAppSignature extends SpCoreSr25519Signature {
                                                                    ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:196:41 - error TS2344: Type '[PolkadotPrimitivesV1DisputeStatement, u32, PolkadotPrimitivesV0ValidatorAppSignature]' does not satisfy the constraint 'AnyTuple'.
  Type 'u32 | PolkadotPrimitivesV0ValidatorAppSignature | PolkadotPrimitivesV1DisputeStatement' is not assignable to type 'Codec'.
    Type 'PolkadotPrimitivesV0ValidatorAppSignature' is missing the following properties from type 'Codec': encodedLength, hash, isEmpty, registry, and 6 more.

196         readonly statements: Vec<ITuple<[PolkadotPrimitivesV1DisputeStatement, u32, PolkadotPrimitivesV0ValidatorAppSignature]>>;
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:221:29 - error TS2304: Cannot find name 'SpCoreEd25519Public'.

221         readonly asEd25519: SpCoreEd25519Public;
                                ~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:223:29 - error TS2304: Cannot find name 'SpCoreSr25519Public'.

223         readonly asSr25519: SpCoreSr25519Public;
                                ~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:311:66 - error TS2344: Type '[u32, PolkadotPrimitivesV0CollatorAppPublic]' does not satisfy the constraint 'AnyTuple'.
  Type 'u32 | PolkadotPrimitivesV0CollatorAppPublic' is not assignable to type 'Codec'.
    Type 'PolkadotPrimitivesV0CollatorAppPublic' is missing the following properties from type 'Codec': encodedLength, hash, isEmpty, registry, and 6 more.

311     interface PolkadotPrimitivesV1ParathreadClaim extends ITuple<[u32, PolkadotPrimitivesV0CollatorAppPublic]> {
                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:330:39 - error TS2344: Type '[PolkadotPrimitivesV0CollatorAppPublic, u32]' does not satisfy the constraint 'AnyTuple'.
  Type 'u32 | PolkadotPrimitivesV0CollatorAppPublic' is not assignable to type 'Codec'.
    Type 'PolkadotPrimitivesV0CollatorAppPublic' is not assignable to type 'Codec'.

330         readonly asParathread: ITuple<[PolkadotPrimitivesV0CollatorAppPublic, u32]>;
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:369:34 - error TS2344: Type 'PolkadotPrimitivesV0ValidatorAppPublic' does not satisfy the constraint 'Codec'.
  Type 'PolkadotPrimitivesV0ValidatorAppPublic' is missing the following properties from type 'Codec': encodedLength, hash, isEmpty, registry, and 6 more.

369         readonly validators: Vec<PolkadotPrimitivesV0ValidatorAppPublic>;
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:370:30 - error TS2344: Type 'PolkadotPrimitivesV0ValidatorAppPublic' does not satisfy the constraint 'Codec'.

370         readonly queued: Vec<PolkadotPrimitivesV0ValidatorAppPublic>;
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:405:34 - error TS2344: Type 'PolkadotPrimitivesV0ValidatorAppPublic' does not satisfy the constraint 'Codec'.

405         readonly validators: Vec<PolkadotPrimitivesV0ValidatorAppPublic>;
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:406:37 - error TS2304: Cannot find name 'SpAuthorityDiscoveryAppPublic'.

406         readonly discoveryKeys: Vec<SpAuthorityDiscoveryAppPublic>;
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@polkadot/types/augment/lookup/types-polkadot.d.ts:407:38 - error TS2344: Type 'PolkadotPrimitivesV1AssignmentAppPublic' does not satisfy the constraint 'Codec'.
  Type 'PolkadotPrimitivesV1AssignmentAppPublic' is missing the following properties from type 'Codec': encodedLength, hash, isEmpty, registry, and 6 more.

407         readonly assignmentKeys: Vec<PolkadotPrimitivesV1AssignmentAppPublic>;
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 34 errors.
```

**Comment by jacogr** [comment]
Duplicate of https://github.com/polkadot-js/api/issues/4213

typegen has not been adapted.

**Comment by jacogr** [comment]
Closed in #4224
